app/proxy.js

```diff
@@ -0,0 +1,58 @@
+var auth = require('./auth');
+var config = require('./config');
+var httpProxy = require('http-proxy');
+var log = require('bole')('proxy');
+
+
+log.info('Starting proxy:', config.proxy);
+var proxy = httpProxy.createProxyServer(config.proxy);
+
+proxy.on('proxyReq', function (proxyReq, req, res, options) {
+  proxyReq.setHeader('Cookie', insert_auth_cookie(req));
+  proxy_body(req, proxyReq);
+});
+
+proxy.on('proxyReqWs', function (proxyReqWs, req, res, options) {
+  proxyReqWs.setHeader('Cookie', insert_auth_cookie(req));
+  proxy_body(req, proxyReq);
+});
+
+function insert_auth_cookie(req) {
+  return insert_cookie(req, 'user-id', auth.cookie(
+    config.rstudio.user, config.rstudio.duration, config.rstudio.key));
+}
+
+function insert_cookie(req, name, value) {
```

Comment:
What is this function doing exactly?

Reply:
It splits the `Cookie` header value by `;` and then filters out any cookie that starts with `{name}=`

Reply:
The point being, it replaces any `user-id=...` cookie with an updated value